### PR TITLE
[TS] LPS-205128 Questions: Accepting an answer under workflow subjects that answer to workflow again

### PR DESCRIPTION
**Pull request by IstvanD**
Dear Team,

Could you please review this pull request?

Motivation
=======
When questions/answers are subject to the built-in Message-Board workflow, the acceptance of an answer by an unprivileged user causes the answer to disappear (assuming: because it has been edited) and to be subject to the workflow again.

Setting a message as an answer will trigger a `patchMessageBoardMessage` mutation API call which will eventually call [MessageBoardMessageResourceImpl._updateMessageBoardMessage](https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/MessageBoardMessageResourceImpl.java#L762C30-L792).

The call to
```
		mbMessage = _mbMessageService.updateDiscussionMessage(
			mbMessage.getClassName(), mbMessage.getClassPK(),
			mbMessage.getMessageId(), headline,
			messageBoardMessage.getArticleBody(),
			_createServiceContext(mbMessage.getGroupId(), messageBoardMessage));
```
will cause a workflow status change on the message.

Proposed Solution
========
In order to avoid the status change, an API method could be introduced that only sets the `answer` field of the `mbMessage`
To avoid passing a boolean as a parameter in the REST URL, I chose to add two methods:

- `putMessageBoardMessageMarkAsAnswer` that sets the `answer` field to `true`
- `putMessageBoardMessageUnmarkAsAnswer` that sets the `answer` field to `false`

Steps to Verify
========
Please see the linked LPS for the reproduction steps.
https://liferay.atlassian.net/browse/LPS-205128

Thank you and best regards,
István

### CODE REVIEW

**Comment by IstvanD** [comment]
Dear Team,

I was wondering if you could find the time to review the pull request.
Please let me know if you need any help or explanation.

Thank you and best regards,
István

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -9824,6 +9824,22 @@ paths:
                             schema:
                                 $ref: "#/components/schemas/MessageBoardMessage"
             tags: ["MessageBoardMessage"]
+    "/message-board-messages/{messageBoardMessageId}/mark-as-answer":
+        put:
```
Comment (adolfopa) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:9828]: A question about the semantics: shouldn't this use POST? This is not putting any subresource, it asking for some processing of the associated message.

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml (hunk 1 above)
Comment (IstvanD) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:9828]: I followed the pattern of the subscribe/unsubscribe action pairs, where an operation on an entry is performed.
https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml#L10056-L10057

The POST requests defined in rest-openapi.yaml are for creating entries, for example:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml#L11380
https://github.com/liferay/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml#L12253

Comment (adolfopa) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml:9828]: Ok, if this is an established pattern them I'm fine with it.

**Comment by adolfopa** [comment]
ci:forward

**Comment by IstvanD** [comment]
Dear Team,

Please let me know if there are any updates on the review.

Thank you and best regards,
István

**Comment by adolfopa** [comment]
@IstvanD The PR is approved; it is on your side to review the test failures and make sure everything is fine.

**Comment by IstvanD** [comment]
ci.test:relevant

**Comment by IstvanD** [comment]
@adolfopa the test failures were not related to the fix. Could you please forward the solution?
Thank you.

**Comment by adolfopa** [comment]
ci:forward:force

**Comment by robertoDiaz** [comment]
ci:forward:force

**Comment by IstvanD** [comment]
Dear Team,

The test failures are not related to the solution.

Best regards,
István

**Comment by adolfopa** [comment]
ci:test:stable

**Comment by adolfopa** [comment]
ci:test:stable

**Comment by IstvanD** [comment]
Resolved the conflicts and rebased the branch.

**Comment by IstvanD** [comment]
Dear Team,

I have checked the test failures are they don't seem to be related to the solution.

If this is the case, could you please forward the solution?

Thank you and best regards,
István

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2532696920

**Comment by AliciaGarciaGarcia** [comment]
ci:report:2532791263

**Pull request closed without merging**